charactermeth.cpp

```diff
@@ -1,38 +1,41 @@
 
-#include <iostream>
 #include "character.h" 
 
-#include <string>
 //a characters oszt�ly met�dusai kifejtve, le�r�s az oszt�lyban
 
-Characters::Characters(std::string name, int Hp, int Dpr) {
-	this->name = name;
-	this->Dpr = Dpr;
-	this->Hp = Hp;
+Characters::Characters(const std::string name,int Hp,const int Dpr) : name(name), Hp(Hp), Dpr(Dpr)
+{	
 }
 
-std::string Characters::Getname() {
+
+
+const std::string& Characters::Getname() {
 	return name;
 }
 
-int Characters::GetHp() {
+const int& Characters::GetHp() {
 	return Hp;
 }
 
-int  Characters::GetDpr() {
+const int&  Characters::GetDpr() {
 	return Dpr;
 }
 
-void Characters::SetHp(int NewHp) {
-	if (NewHp < 0) {
+void Characters::SetHp(Characters &X) {
+
+	
+	if (Hp - X.GetDpr() < 0) {
 		Hp = 0;
 	}
 	else {
-		Hp = NewHp;
+		Hp = Hp - X.GetDpr();
 	}
+	

```

Comment:
Na így már jobb, de akkor ne az legyen a fgv neve, hogy SetHp, meg a headerben is írjátok át a dokumentációját. És mivel a paraméterben kapott támadónak nem szabad módosulnia a támadás során, a paraméter legyen const!